rust/agama-server/src/network/model.rs

```diff
@@ -1313,6 +1314,25 @@ impl fmt::Display for InfinibandTransportMode {
     }
 }
 
+
+#[derive(Default, Debug, PartialEq, Clone, Serialize)]
+pub enum TunMode {
+    #[default]
+    Tun = 1,
+    Tap = 2,
+}
+
+#[derive(Default, Debug, PartialEq, Clone, Serialize)]
+pub struct TunConfig {
+    pub mode: TunMode,
+    pub group: Option<String>,
+    pub owner: Option<String>,
+    pub multi_queue: bool,
```

Comment:
I'm not very used to tunneling terminology and protocol, and specially this variable is not self descriptive to me, in fact in the NM Device Settings it is NoPi (no protocol information), apart of that, which is the wicked equivalent name?, it is also configurable in wicked?

Reply:
Wicked only support mode, group and owner. Everything else was added because of NetworkManager.

I would say multi_queue is this: `From version 3.8, Linux supports multiqueue tuntap which can uses multiple
file descriptors` (see [kernel.org/tuntap.txt](https://www.kernel.org/doc/Documentation/networking/tuntap.txt) )
And the same as you can configure with `ip`
```
# ip tuntap help
Usage: ip tuntap { add | del | show | list | lst | help } [ dev PHYS_DEV ]
	[ mode { tun | tap } ] [ user USER ] [ group GROUP ]
	[ one_queue ] [ pi ] [ vnet_hdr ] [ multi_queue ] [ name NAME ]
```

Reply:
The thing is ... if we do not need it by now.. shall we add it to our model?

Reply:
Ok I agree, we will omit them for now.